keys_test.go

```diff
@@ -1,23 +1,233 @@
 package crypto11
 
 import (
+	"crypto/rand"
+	"crypto/rsa"
+	"github.com/miekg/pkcs11"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func TestFindKeysRequiresIdOrLabel(t *testing.T) {
+// withContext executes a test function with a context.
+func withContext(t *testing.T, f func(ctx *Context)) {
 	ctx, err := ConfigureFromFile("config")
 	require.NoError(t, err)
 
 	defer func() {
 		require.NoError(t, ctx.Close())
 	}()
 
-	_, err = ctx.FindKey(nil, nil)
-	assert.Error(t, err)
+	f(ctx)
+}
+
+func TestFindKeysRequiresIdOrLabel(t *testing.T) {
+	withContext(t, func(ctx *Context) {
+		_, err := ctx.FindKey(nil, nil)
+		assert.Error(t, err)
+
+		_, err = ctx.FindKeys(nil, nil)
+		assert.Error(t, err)
+
+		_, err = ctx.FindKeyPair(nil, nil)
+		assert.Error(t, err)
+
+		_, err = ctx.FindKeyPairs(nil, nil)
+		assert.Error(t, err)
+	})
+}
+
+func TestFindingKeysWithAttributes(t *testing.T) {
+	withContext(t, func(ctx *Context) {
+		id := randomBytes()
+		id2 := randomBytes()
+
+		key, err := ctx.GenerateSecretKey(id, 128, CipherAES)
+		require.NoError(t, err)
+		require.NotNil(t, key)
```

Comment:
Why the `nil` check? I know some other tests have it, but I'm sure it's unneeded.

Reply:
I'll remove them then